src/rhsm/logutil.py

```diff
@@ -16,6 +16,15 @@
 import os
 import sys
 import rhsm.config
+import datetime
+
+
+class RhsmISO8601Formatter(logging.Formatter):
+    """Ensure date & time to be in ISO8601 format and containing info about milliseconds and time zone"""
+
+    def formatTime(self, record, datefmt=None):
+        return datetime.datetime.fromtimestamp(record.created).astimezone().isoformat(timespec="milliseconds")
```

Comment:
the place of this new class in the file is a bit off, i.e. in the middle of the module-level variables; I'd recommend to place it after the `PyWarningsLogger` class